sound/soc/sof/intel/hda-dsp.c

```diff
@@ -406,6 +409,11 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 	}
 
 	reg = snd_sof_dsp_read8(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset);
+	if ((reg & SOF_HDA_VS_D0I3C_I3) != value) {
+		dev_err(sdev->dev, "failed to update D0I3C!\n");
+		return -EIO;
+	}
+
```

Comment:
is this useful?

````c
static int hda_dsp_wait_d0i3c_done(struct snd_sof_dev *sdev)
{
	int retry = HDA_DSP_REG_POLL_RETRY_COUNT;
	struct snd_sof_pdata *pdata = sdev->pdata;
	const struct sof_intel_dsp_desc *chip;

	chip = get_chip_info(pdata);
	while (snd_sof_dsp_read8(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset) &
		SOF_HDA_VS_D0I3C_CIP) {
		if (!retry--)
			return -ETIMEDOUT;
		usleep_range(10, 15);
	}

	return 0;
}
````

we've already validated the value above and return -ETIMEOUT.

Reply:
@plbossart the check for cip only checks if the command went through or not. It doesn't check if the desired value is set or not.

Reply:
@ranj063 well in that case, isn't the implementation of hda_dsp_wait_d0i3c_done() already wrong, in that it doesn't check if the desired value is set.

My point is that doing two things differently or twice looks like code that needs attention.

Reply:
maybe its the name hda_dsp_wait_d0i3c_done() that's misleading. It should be called hda_dsp_check for_d0i3c_cip() instead? In any case, maybe we need to disconnect the 2 patches and move the D0i3 stuff to a new PR

Reply:
I am not sure what this function is supposed to do and why we care about the value in one case and not the other.

Should be similar to SPA/CPA in that both enable/disable check that the desired value is reached.

Reply:
While cip is set, the other bits in d0i3 register are invalid and also it is illegal for sw to write to any bit in this register. This function is used to check cip bit in d0i3 to make sure it is valid to read or write.  This function is fine to me